src/service/viewer-impl.js

```diff
@@ -445,7 +452,40 @@ export class Viewer {
    * @return {!VisibilityState}
    */
   getVisibilityState() {
-    return this.visibilityState_;
+    // The viewer has informed us that the doc is no longer active, meaning
```

Comment:
Let's store a single and correct `this.visibilityState_` (point of reference) and recalculate it only when something changes.